projects/step-frontend/src/lib/modules/timeseries/components/dashboard/dashboard.component.ts

```diff
@@ -220,7 +232,7 @@ export class DashboardComponent implements OnInit, OnDestroy {
       .onTimeSelectionChange()
       .pipe(
         switchMap((newRange) => this.handleSelectionChange(newRange)),
-        takeUntil(compareCharts ? this.terminator$ : this.terminator$)
+        takeUntil(compareCharts ? this.terminator$ : this.terminator$),
```

Comment:
Very strange condition. I think it should be `takeUntil(this.terminator$)`

Also in current version of angular instead of terminator$, the following approach can be used:
```
private _destroyRef = inject(DestroyRef);

private someMethod(): void {
   ...
   .pipe(
        ...,
        takeUntilDestroyed(this._destroyRef)
   )
   .subscribe(() => {
        ....
   })
}

```
The difference in comparison with terminator subject, that there is no need to define additional `ngOnDestroy` hook and call next/complete for the subject.

Reply:
@dvladir good point!